### PR TITLE
Feature/efrs 975 add new frames

### PR DESCRIPTION
**Pull request by pbohdan**

### CODE REVIEW

File: ui/src/app/store/face-verification/reducers.ts [hunk 1]
```diff
@@ -34,8 +43,14 @@ const initialStateVerification: FaceVerificationEntityState = {
 
 const reducerVerification: ActionReducer<FaceVerificationEntityState> = createReducer(
   initialStateVerification,
-  on(verifyFace, verifyFaceSuccess, (state, action) => ({ ...state, ...action, isPending: false })),
-  on(verifyFaceReset, verifyFaceFail, () => ({ ...initialStateVerification }))
+  on(verifyFaceAddProcessFile, (state, action) => ({ ...state, ...action })),
+  on(verifyFaceAddCheckFileFile, (state, action) => ({ ...state, ...action })),
```
Comment (isobolev) [inline ui/src/app/store/face-verification/reducers.ts:47]: please combine multiple actions in 1 reducer function

File: ui/src/app/store/face-verification/reducers.ts [hunk 2]
```diff
@@ -34,8 +43,14 @@ const initialStateVerification: FaceVerificationEntityState = {
...
+  on(verifyFace, state => ({ ...state, isPending: true })),
+  on(verifyFaceSuccess, (state, action) => ({ ...state, ...action, isPending: false })),
+  on(verifyFaceFail, state => ({ ...state, isPending: false })),
+  on(verifyFaceProcessFileReset, verifyFaceFail, state => ({ ...state, processFile: null, request: null, model: null })),
+  on(verifyFaceCheckFileReset, state => ({ ...state, checkFile: null, request: null, model: null })),
```
Comment (isobolev) [inline ui/src/app/store/face-verification/reducers.ts:52]: please combine multiple actions in 1 reducer function

File: ui/src/app/data/interfaces/response-result.ts [hunk 3]
```diff
@@ -15,48 +15,43 @@
  */
 
 /* eslint-disable @typescript-eslint/naming-convention */
+export interface BoxSize {
```
Comment (isobolev) [inline ui/src/app/data/interfaces/response-result.ts:18]: 1 file = 1 interface. Interface name === file name.

File: ui/src/app/features/face-services/face-picture/face-picture.component.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  @ViewChild('canvasPicture', { static: true }) canvasPicture: ElementRef<HTMLCanvasElement>;
+  @ViewChild('canvasLandmarks', { static: true }) canvasLandmarks: ElementRef<HTMLCanvasElement>;
+
+  types = ServiceTypes;
+  disableLandmarks = false;
+
+  private currentPicture: () => any;
+  private colorLandmarks = '#27C224';
```
Comment (isobolev) [inline ui/src/app/features/face-services/face-picture/face-picture.component.ts:44]: harcoded styles?

File: ui/src/app/features/face-services/face-picture/face-picture.component.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    ctx.fillStyle = color;
+
+    data.forEach(val => this.displayLandmarks(ctx, val, color));
+  }
+
+  displayLandmarks(ctx: CanvasRenderingContext2D, data: RequestResultRecognition | FaceMatches | SourceImageFace, color: string): void {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { x_max, x_min, y_max, y_min } = data.box;
+    const frameArea: number = Math.round(x_max - x_min) * Math.round(y_max - y_min);
+    const sizePoint = frameArea / 3000 / 10 > 1.6 || data.landmarks.length >= 108 ? frameArea / 3000 / 10 : 1.6;
```
Comment (isobolev) [inline ui/src/app/features/face-services/face-picture/face-picture.component.ts:93]: what is 3000? worth to extract to constant?

**Review by isobolev — COMMENTED** [review]
I saw this in action, and it looks really cool, just a few minor comments. Nice job Pavel!

File: ui/src/app/store/face-verification/reducers.ts (hunk 1 above)
Comment (pbohdan) [reply ui/src/app/store/face-verification/reducers.ts:47]: fixed

File: ui/src/app/data/interfaces/response-result.ts (hunk 3 above)
Comment (pbohdan) [reply ui/src/app/data/interfaces/response-result.ts:18]: fixed

File: ui/src/app/store/face-verification/reducers.ts (hunk 2 above)
Comment (pbohdan) [reply ui/src/app/store/face-verification/reducers.ts:52]: fixed

File: ui/src/app/features/face-services/face-picture/face-picture.component.ts (hunk 4 above)
Comment (pbohdan) [reply ui/src/app/features/face-services/face-picture/face-picture.component.ts:44]: fixed

**Review by isobolev — APPROVED** [review]

**Pull request merged**